subway_prediction_ml/test/test_locations_adder.py

```diff
@@ -33,3 +33,31 @@ def test_n_hot_train_locations_added_to_vehicle_datapoints(self):
             ['B-112', 12345, "44--72", 1.0, 0.0, 1.0, 0.0],
             ['B-112', 12346, "45--73", 0.0, 1.0, 0.0, 1.0],
         ]
+
+    def test_locations_can_be_filtered_by_line(self):
+        locations_data = pd.DataFrame(
+            columns=["loc_id", "line"],
+            data=[
+                ["42--70", "R"],
+                ["43--71", "O"],
+                ["44--72", "R"],
+                ["45--73", "G"]
+            ]
+        )
+
+        vehicle_data = pd.DataFrame(
+            columns=['trip_id', 'generation', 'current_location_id'],
+            data=[
+                ['B-111', 12345, "42--70"],
+                ['B-112', 12345, "44--72"],
+                ['B-111', 12347, "44--72"]
```

Comment:
I'm not sure that this test really exercises the location filtering, since none of the vehicles would be joined with any non-"R" locations anyway. Can you add a datapoint to this `vehicle_data` data frame for `"43-71"` (orange line, above). And then we would expect to see that data point filtered out after the `fit_transform`, I think.

Reply:
I thought so too, but if you try putting in non-Red vehicle datapoints, it doesn't filter them out, but just appends the n-hotted Red Line location columns like usual. Never mind that these datapoints are on a different line. By the time the vehicle datapoints reach this stage of the pipeline, they should have been filtered by route already if they were going to be, so I think this is OK.

Reply:
Ah, right. So the filtering is actually captured by this test, since the 1-hotted locations in the assertion only have two columns.